### PR TITLE
Add user agent header

### LINKED ISSUE #104: Add User Agent Header
Add the user-agent request header to every call in this sdk.

Example:

headers: {
   Authorization: `Bearer ${ApiKey}`,
   "User-Agent": `novu/${sdk}@${version}`,
},

Why? (Context)

Having request headers in the SDKs will help us in monitoring which SDK is mostly used by the community and customers. It will help us make the decisions in the future on which SDKs to officially support in the future.

### PR DESCRIPTION
**Pull request by VishalMCF**
resolves #104

### CODE REVIEW

**Comment by VishalMCF** [comment]
@Cliftonz  Can you please review?

**Comment by unicodeveloper** [comment]
@mayorJAY Please review

**Review by Cliftonz — APPROVED** [review]

**Comment by mayorJAY** [comment]
This implementation would give us the name and version of the client app instead of this SDK. We need to revert and revisit the implementation @Cliftonz @unicodeveloper @VishalMCF

File: src/main/java/co/novu/common/base/NovuConfig.java [hunk 1]
```diff
@@ -1,16 +1,34 @@
...
         this.apiKey = apiKey;
+        loadContextFromPom();
     }
 
-    private String apiKey;
-    private String baseUrl = "https://api.novu.co/v1/";
+    private void loadContextFromPom(){
+        try {
+            MavenXpp3Reader reader = new MavenXpp3Reader();
+            Model model = reader.read(new FileReader("pom.xml"));
```
Comment (mayorJAY) [inline src/main/java/co/novu/common/base/NovuConfig.java:27]: Reading pom.xml directly will result in reading the pom file of the client app. Please do this instead:
`Model model = reader.read(new InputStreamReader(Objects.requireNonNull(this.getClass().getResourceAsStream("/META-INF/maven/co.novu/novu-java/pom.xml"))));`

File: src/main/java/co/novu/common/base/NovuConfig.java [hunk 2]
```diff
@@ -1,16 +1,34 @@
...
 import lombok.Setter;
 
 @Getter
 @Setter
 public class NovuConfig {
 
+    private String apiKey;
+    private String baseUrl = "https://api.novu.co/v1/";
+    private String sdkName;
+    private String sdkVersion;
```
Comment (mayorJAY) [inline src/main/java/co/novu/common/base/NovuConfig.java:18]: Having these fields here will give the users the liberty to provide the name and version which shouldn't be. Please move this implementation into the RestHandler.java class.

By doing that, you don't need these fields, just have a function that returns the version, the name can be hardcoded, it should be **Java**

**Comment by VishalMCF** [comment]
@mayorJAY I was thinking of an approach where I hardcode the version in the RetryHandlerClass. Name as you said can be hardcoded. I can write a test function that always checks if the version is the same as what we have in the pom.xml file. What do you say?

**Comment by mayorJAY** [comment]
> @mayorJAY @Cliftonz @unicodeveloper I was thinking of an approach where I hardcode the version in the RetryHandlerClass. Name as you said can be hardcoded. I can write a test function that always checks if the version is the same as what we have in the pom.xml file. What do you say? That way don't have to risk reading the file during runtime.

I don't think we should hardcode the version, that would mean updating the header every time there is a version change. If we modify the code based on my suggestions, we'd get what we need

**Comment by VishalMCF** [comment]
@mayorJAY cool, I will raise a new PR as per your suggestions

**Pull request merged**